Menus/TitleScreen.cpp

```diff
@@ -0,0 +1,608 @@
+		m_FmodLogo = nullptr;
+		m_PreGameLogoText.Reset();
+		m_PreGameLogoTextGlow.Reset();
+
+		m_IntroSlides.fill(nullptr);
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void TitleScreen::Create(AllegroScreen *guiScreen) {
+		CreateTitleElements();
+
+		m_IntroScrollStartOffsetY = (static_cast<float>(m_Nebula.GetBitmap()->h) / m_BackdropScrollRatio) - (static_cast<float>(g_FrameMan.GetResY()) / m_BackdropScrollRatio);
+		m_ScrollOffset.SetXY(0, m_IntroScrollStartOffsetY);
+
+		m_StationOrbitTimer.SetElapsedRealTimeS(15);
+
+		if (!g_FrameMan.ResolutionChanged()) {
+			if (!g_SettingsMan.SkipIntro() && !g_ActivityMan.LaunchIntoActivity()) {
+				m_IntroTextFont = std::make_unique<GUIFont>("FontLarge");
+				m_IntroTextFont->Load(guiScreen, "Base.rte/GUIs/Skins/Menus/FontLarge.png");
+				CreateIntroSequenceSlides();
+			} else {
+				m_IntroSequenceState = IntroSequence::MainMenuAppear;
+			}
+		} else {
+			SetTitleTransitionState(TitleTransition::MainMenu);
+			m_FinishedPlayingIntro = true;
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void TitleScreen::CreateTitleElements() {
+		m_DataRealmsLogo = ContentFile("Base.rte/GUIs/Title/Intro/DRLogo5x.png").GetAsBitmap(COLORCONV_NONE, false);
+		m_FmodLogo = ContentFile("Base.rte/GUIs/Title/Intro/FMODLogo.png").GetAsBitmap(COLORCONV_NONE, false);
+
+		m_PreGameLogoText.Create(ContentFile("Base.rte/GUIs/Title/Intro/PreTitle.png"));
+		m_PreGameLogoTextGlow.Create(ContentFile("Base.rte/GUIs/Title/Intro/PreTitleGlow.png"));
+		m_GameLogo.Create(ContentFile("Base.rte/GUIs/Title/Title.png"));
+		m_GameLogoGlow.Create(ContentFile("Base.rte/GUIs/Title/TitleGlow.png"));
+		m_Planet.Create(ContentFile("Base.rte/GUIs/Title/Planet.png"));
+		m_Moon.Create(ContentFile("Base.rte/GUIs/Title/Moon.png"));
+		m_Station.Create(ContentFile("Base.rte/GUIs/Title/Station.png"));
+		m_Nebula.Create(ContentFile("Base.rte/GUIs/Title/Nebula.png"), false, Vector(), false, false, Vector(0, -1.0F));
+
+		set_write_alpha_blender();
+		draw_trans_sprite(m_PreGameLogoText.GetSpriteFrame(0), ContentFile("Base.rte/GUIs/Title/Intro/PreTitleAlpha.png").GetAsBitmap(COLORCONV_NONE, false), 0, 0);
+		draw_trans_sprite(m_GameLogo.GetSpriteFrame(0), ContentFile("Base.rte/GUIs/Title/TitleAlpha.png").GetAsBitmap(COLORCONV_NONE, false), 0, 0);
+		draw_trans_sprite(m_Planet.GetSpriteFrame(0), ContentFile("Base.rte/GUIs/Title/PlanetAlpha.png").GetAsBitmap(COLORCONV_NONE, false), 0, 0);
+		draw_trans_sprite(m_Moon.GetSpriteFrame(0), ContentFile("Base.rte/GUIs/Title/MoonAlpha.png").GetAsBitmap(COLORCONV_NONE, false), 0, 0);
+
+		int starSmallBitmapCount = 4;
+		std::vector<BITMAP *> starSmallBitmaps = ConvertDoublePointerToVectorOfPointers(ContentFile("Base.rte/GUIs/Title/Stars/StarSmall.png").GetAsAnimation(starSmallBitmapCount), starSmallBitmapCount);
```

Comment:
I should have thought of this when you first mentioned the double pointer thing a while back, but why not just add some alternate GetAsAnimation method that gives you the vector directly? It'll hide away a bit of the complexity and make things look cleaner.

Reply:
We'll deal with eliminating devil double pointers some other time.